examples/apps/resizable-inline/src/main.rs

```diff
@@ -0,0 +1,266 @@
+/// A Ratatui example that demonstrates how to use the resizable inline viewport.
+///
+/// This example shows how to create an inline viewport with a fixed height that can be
+/// dynamically resized during runtime using the new `set_viewport_height()` method.
+///
+/// Controls:
+/// - '+' or '=' : Increase viewport height
+/// - '-' or '_' : Decrease viewport height  
+/// - 'r'        : Reset viewport to initial height
+/// - 'i'        : Insert a line before the viewport
+/// - 'q'        : Quit
+///
+/// This example runs with the Ratatui library code in the branch that you are currently
+/// reading. See the [`latest`] branch for the code which works with the most recent Ratatui
+/// release.
+///
+/// [`latest`]: https://github.com/ratatui/ratatui/tree/latest
+use std::{
+    sync::mpsc,
+    thread,
+    time::{Duration, Instant},
+};
+
+use color_eyre::Result;
+use crossterm::event::{self, KeyCode, KeyEventKind};
+use ratatui::backend::Backend;
+use ratatui::layout::{Constraint, Layout};
+use ratatui::style::{Color, Modifier, Style};
+use ratatui::text::{Line, Span};
+use ratatui::widgets::{Block, Borders, Gauge, List, ListItem, Paragraph, Widget};
+use ratatui::{Frame, Terminal, TerminalOptions, Viewport};
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+    
+    // Start with an inline viewport of height 6
+    let mut terminal = ratatui::init_with_options(TerminalOptions {
+        viewport: Viewport::Inline(6),
+    });
+
+    let (tx, rx) = mpsc::channel();
+    input_handling(tx.clone());
+    
+    let mut app_state = AppState::new();
+    let app_result = run(&mut terminal, &mut app_state, rx);
+
+    ratatui::restore();
+    app_result
+}
+
+struct AppState {
+    current_height: u16,
```

Comment:
nit: every variable in a program is "current" unless otherwise mentioned
```suggestion
    height: u16,
```